Tests/SentryProfilerTests/SentryProfilerSwiftTests.swift

```diff
@@ -207,121 +203,92 @@ class SentryProfilerSwiftTests: XCTestCase {
 
     private var fixture: Fixture!
 
+    override class func setUp() {
+        super.setUp()
+        SentryLog.configure(true, diagnosticLevel: .debug)
+    }
+
     override func setUp() {
         super.setUp()
         fixture = Fixture()
-        SentryLog.configure(true, diagnosticLevel: .debug)
-        CurrentDate.setCurrentDateProvider(fixture.currentDateProvider)
-        fixture.options.profilesSampleRate = 1.0
-        fixture.options.tracesSampleRate = 1.0
-        fixture.prepareMetricsMocks()
     }
 
     override func tearDown() {
         super.tearDown()
+        fixture.timeoutTimerWrapper.fire()
         clearTestState()
     }
 
     func testMetricProfiler() throws {
-        // start span
         let span = try fixture.newTransaction()
         forceProfilerSample()
-
         try fixture.gatherMockedMetrics(span: span)
-
-        // finish profile
-        let exp = expectation(description: "Receives profile payload")
-        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
-            span.finish()
-            do {
-                try self.assertMetricsPayload()
-                exp.fulfill()
-            } catch {
-                XCTFail("Encountered error: \(error)")
-            }
-        }
-        waitForExpectations(timeout: 3)
+        self.fixture.currentDateProvider.advanceBy(nanoseconds: 1.toNanoSeconds())
+        span.finish()
+        try self.assertMetricsPayload()
     }
 
     func testConcurrentProfilingTransactions() throws {
```

Comment:
This test case is the only one with changes pertinent to the "current" vs "previous" timestamp change, but mostly they involve correctly setting up mock state from how it was previously done. 

The test already had the following structure:

1. set up 10 spans
2. make sure each one is reported with the correct metrics/samples sliced to tx bounds
3. reset everything
4. do it all again to make sure stopping and starting the profiler again doesn't break anything (basically, copy/paste 1 and 2)

The other change in this test case is that the bodies of the loops between 1/2 and 4 had slightly diverged; extract them to a common function so they run exactly the same way both times.